core/src/main/java/org/apache/gravitino/GravitinoEnv.java

```diff
@@ -226,6 +226,26 @@ public IdGenerator idGenerator() {
     return idGenerator;
   }
 
+  /**
+   * Get the CatalogManager associated with the Gravitino environment.
+   *
+   * @return The CatalogManager instance.
+   */
+  public CatalogManager catalogManager() {
+    Preconditions.checkArgument(catalogManager != null, "GravitinoEnv is not initialized.");
```

Comment:
Is it necessary to do the check?

Reply:
This variable is lazily initialized, we need to ensure it has been initialized when accessed.

Reply:
could you provide the scene `catalogManager` is not initialized when accessed? could it really happen?

Reply:
Because it is lazy-loaded, it can happen anywhere.